plugins/net_plugin/net_plugin.cpp

```diff
@@ -4364,23 +4364,23 @@ namespace eosio {
       for(auto listen_itr = listen_addresses.begin(), p2p_iter = p2p_addresses.begin();
           listen_itr != listen_addresses.end();
           ++listen_itr, ++p2p_iter) {
-         app().executor().post(priority::highest, [my=shared_from_this(), address = std::move(*listen_itr), p2p_addr = *p2p_iter](){
+         boost::asio::post(thread_pool.get_executor(), [this, address = std::move(*listen_itr), p2p_addr = *p2p_iter]() {
```

Comment:
In the original code the main thread was posting to the app thread presumably to delay the listeners from being started until the main thread event loop starts. With this change, that is no longer the case. The listeners (and also the timers and outgoing connections) will be started immediately. Does that change anything? Does it affect anything wrt replay?

Reply:
`net_plugin` APPBASE_PLUGIN_REQUIRES `chain_plugin` and `producer_plugin`. Replay will be finished before `net_plugin::plugin_startup` is called. There should be no issue with accepting trxs and blocks before the main thread is started.

Reply:
What was the purpose of the main thread to main thread post that was previously there? Delaying startup until main thread event loop is running is the only rationale I'm coming up with

Reply:
Speed up startup maybe. It might have been done originally because of some startup issue, but doesn't seem necessary.

Reply:
Adding that main thread to main thread post was described expressly to fix replay: 925007b (from eosnetworkfoundation/mandel#381).

Reply:
I'm not sure what I was thinking there. The fix there is the reconnect on sync timeout. I've testing the current version of this PR with create_listener on the main thread at startup. I tried a number of different replays and found no issue with listening or connecting after the replay finishes.